android/src/main/java/com/oney/WebRTCModule/GetUserMediaImpl.java

```diff
@@ -233,14 +234,43 @@ void disposeTrack(String id) {
         }
     }
 
-    void switchCamera(String trackId) {
+    void switchCamera(String trackId, Promise promise) {
         TrackPrivate track = tracks.get(trackId);
         if (track != null && track.videoCaptureController instanceof CameraCaptureController) {
             CameraCaptureController cameraCaptureController = (CameraCaptureController) track.videoCaptureController;
-            cameraCaptureController.switchCamera();
+            cameraCaptureController.switchCamera(new Consumer<Exception>() {
+                public void accept(Exception e) {
+                    if(e != null) {
+                        promise.reject(e);
+                        return;
+                    }
+
+                    promise.resolve(cameraCaptureController.getSettings());
+                }
+            });
+        } else {
+            promise.reject(new Exception("Camera track not found!"));
         }
     }
 
+    void applyConstraints(String trackId, ReadableMap constraints, Promise promise) {
+        TrackPrivate track = tracks.get(trackId);
+        if (track != null && track.videoCaptureController instanceof CameraCaptureController) {
```

Comment:
I guess we might need to tweak this in the future if screen tracks can be tweaked too. Would it make sense to have an implementation in the base class that always returns an error and override it in the camera capturer?